### PR TITLE
Support for adding a single label classifier to a pipeline

### PR DESCRIPTION
**Pull request by AlanAboudib**
## Description
This PR implements the `SingleLabelClassifier` class which uses an `AverageDocEncoder`.

It also supports adding the `SingleLabelClassifier` to the pipeline which could then be deployed to the Grid and used for inference on strings situated on any worker.

In this PR, the classifier is downloaded to the worker were data is situated. In a future PR, remote inference where the classifier is not downloaded to the data owner's worker is to be supported.

Fixes: #134 #135 #136 #137 

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Tested locally. But CI test to be created

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: syfertext/language.py [hunk 1]
```diff
@@ -22,6 +23,9 @@
 from typing import Set
 from typing import Dict
 
+from collections import defaultdict
+import numpy as np
```
Comment (Nilanshrajput) [inline syfertext/language.py:27]: numpy is not used anywhere previously,I converted it to torch tensor everywhere

File: syfertext/language.py [hunk 2]
```diff
@@ -90,10 +96,8 @@ def set_tokenizer(self, tokenizer: Tokenizer, access: Set[str] = None):
         if access is None:
             access = {self.owner.id}
 
-        # The name of the tokenizer component is set to the class name
-        # for the moment
-        name = tokenizer.__class__.__name__.lower()
-        self.tokenizer = tokenizer
+        # Set the name of the tokenizer
+        name = tokenizer.__class__.__name__.lower() if name is None else name
```
Comment (Nilanshrajput) [inline syfertext/language.py:100]:
```suggestion
        if name is None:
            name = tokenizer.__class__.__name__.lower() if name is None else name
```

File: syfertext/pipeline/average_doc_encoder.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        """
+
+        # Case 1: SMPC encryption required.
+        # In this case, `doc` is a DocPointer object although that it
+        # could also be a Doc object in theory.
+        # The returned tensor is a Syft AdditiveSharingTensor object.
+        if "mpc" in crypto_config:
+
+            # Encrypt the vector
+            vector_enc = doc.get_encrypted_vector(
```
Comment (Nilanshrajput) [inline syfertext/pipeline/average_doc_encoder.py:95]: mpc could be done using more than 3 nodes also I think, I think better config will be like {'mpc':List[nodes]} and pass all the nodes in the list here.

File: syfertext/language.py (hunk 1 above)
Comment (AlanAboudib) [reply syfertext/language.py]: Done

**Pull request merged**